Source/player.cpp

```diff
@@ -147,11 +147,11 @@ bool PlrDirOK(int pnum, Direction dir)
 	}
 
 	if (dir == DIR_E) {
-		return !SolidLoc(position + DIR_SE) && (dFlags[position.x + 1][position.y] & BFLAG_PLAYERLR) == 0;
+		return !IsTileSolid(position + DIR_SE) && (dFlags[position.x + 1][position.y] & BFLAG_PLAYERLR) == 0;

```

Comment:
```suggestion
		return IsTileNotSolid(position + DIR_SE) && (dFlags[position.x + 1][position.y] & BFLAG_PLAYERLR) == 0;
```
`!IsTileSolid` is true for out of bounds positions. This is a pre-existing bug that theoretically could lead to OOB reads with the dFlags check.